src/llm_tracekit/bedrock/invoke_agent.py

```diff
@@ -28,13 +32,94 @@
 from llm_tracekit.span_builder import (
     Choice,
     Message,
+    ToolCall,
     attribute_generator,
     generate_base_attributes,
     generate_choice_attributes,
     generate_message_attributes,
     generate_response_attributes,
 )
 
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class AgentStreamResult:
+    """
+    A data transfer object to hold all the collected results from the agent stream.
+    This avoids passing a long list of parameters between functions.
+    """
+
+    content: Optional[str]
+    usage_input_tokens: Optional[int]
+    usage_output_tokens: Optional[int]
+    foundation_model: Optional[str]
+    inference_config_max_tokens: Optional[int]
+    inference_config_temperature: Optional[float]
+    inference_config_top_k: Optional[int]
+    inference_config_top_p: Optional[float]
+    # Chat history is a tuple containing the list of prompts and the list of completions.
+    chat_history: Optional[Tuple[List[Message], List[Choice]]]
+    # A buffer for tool calls that have been observed but not yet assigned to an assistant's choice.
+    tool_calls_buffer: List[ToolCall] = field(default_factory=list)
+
+
+class BedrockMessageParser:
```

Comment:
The class is redundant - if you want to separate this logic from the rest of the module put it in a helper module and import the functions you need